app/routes/quiz.rb

```diff
@@ -8,7 +8,18 @@ class Flippd < Sinatra::Application
   post '/quiz/:id/mark' do
     # Get the answers from the form
     @answers = extract_post_data(params)
-    @results = Quiz.get(params[:id]).mark(@answers)
+    quiz = Quiz.get(params[:id])
+    @results = quiz.mark(@answers)
+    # Only save if logged in.
+    if @user then
+
+      quiz = Quiz.get(params[:id])
```

Comment:
No worries. I'll remove this line and simplify the loop in routes/video.rb then update the PR.